MdePkg/Include/Register/Intel/StmApi.h

```diff
@@ -50,12 +53,25 @@ typedef struct {
   ///
   /// The total STM_HEADER should be 4K.
   ///
+  
+  // Pad to take up 1KB of space
+  UINT8     Reserved2[SIZE_1KB - 5 * sizeof (UINT32) - 3 * sizeof (UINT8) - sizeof (UINT16) - sizeof (STM_FEAT)];

```

Comment:
I think the best way to deal with these is to statically verify them against the compiler generated value.

For example:

```c
#define SOFTWARE_STM_HEADER_PADDING_SIZE (SIZE_1KB - 5 * sizeof (UINT32) - 3 * sizeof (UINT8) - sizeof (UINT16) - sizeof (STM_FEAT))

// ... Later in the struct definition...

  ///
  /// The total STM_HEADER should be 4K.
  ///

  // Pad to take up 1KB of space
  UINT8     Reserved2[SOFTWARE_STM_HEADER_PADDING_SIZE];
} SOFTWARE_STM_HEADER;

// Afterward, confirm...

STATIC_ASSEERT ((SIZE_1KB - OFFSET_OF (SOFTWARE_STM_HEADER, Reserved2) == SOFTWARE_STM_HEADER_PADDING_SIZE, "SOFTWARE_STM_HEADER_PADDING_SIZE mismatch!");

// Continue on...
```

Reply:
Would you want to make sure the 3 fields are 2KB, 1KB, and 1KB? Just to avoid touching too many details of the members inside those sub-structures?

Reply:
I think that makes sense to leave the original type definitions matching the upstream.

Reply:
Good idea.  I moved the padding calculations to defines as suggested and now they're being validated with STATIC_ASSERTs.